### PR TITLE
OAK-9714 | Adding support to be able to build FlatFileStore independe…

### PR DESCRIPTION
**Pull request by nit0906**
…nt of the index method in DocumentStoreIndexer

### CODE REVIEW

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java [hunk 1]
```diff
@@ -69,6 +74,7 @@
     private final IndexEditorProvider indexEditorProvider;
     private final AsyncIndexerLock indexerLock;
     private final IndexDefinitionUpdater indexDefinitionUpdater;
+    private final boolean preserveCheckpoint = Boolean.parseBoolean(System.getProperty(OAK_INDEX_IMPORTER_PRESERVE_CHECKPOINT, "false"));
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java]: I think it is equivalent to Boolean.getBoolean(x): https://docs.oracle.com/javase/9/docs/api/java/lang/Boolean.html#getBoolean-java.lang.String-

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java [hunk 2]
```diff
@@ -290,8 +296,14 @@ static String getAsyncLaneName(String indexPath, NodeState indexState) {
     }
 
     private void releaseCheckpoint() {
-        nodeStore.release(indexerInfo.checkpoint);
-        log.info("Released the referred checkpoint [{}]", indexerInfo.checkpoint);
+        if (!preserveCheckpoint) {
```
Comment (thomasmueller) [inline oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java]: Nit: I would use "if (preserveCheckpoint)" and then swap the "then" and "else" parts.

File: oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStore.java [hunk 3]
```diff
@@ -50,6 +50,16 @@ public FlatFileStore(BlobStore blobStore, File storeFile, NodeStateEntryReader e
...
     }
 
+    public String getFlatFileStorePath() {
+        if (storeFile.exists() && storeFile.isFile() && storeFile.canRead()) {
+            return storeFile.getAbsolutePath();
+        }
+
+        String msg = String.format("Cannot read store file at [%s]",
+                storeFile.getAbsolutePath());
+        throw new IllegalArgumentException(msg);
```
Comment (thomasmueller) [inline oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStore.java]: I wonder if we could move the exception handling to the setter... Having it in the getter is weird in my view.

**Review by thomasmueller — CHANGES_REQUESTED** [review]

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java (hunk 1 above)
Comment (nit0906) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java]: done.

File: oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java (hunk 2 above)
Comment (nit0906) [reply oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/importer/IndexImporter.java]: done

File: oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStore.java (hunk 3 above)
Comment (nit0906) [reply oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStore.java]: done

**Comment by nit0906** [comment]
@thomasmueller  - made the changes as per suggestions. Please re-review.

**Review by thomasmueller — APPROVED** [review]

**Review by tihom88 — APPROVED** [review]

**Pull request merged**